### PR TITLE
Phase2-hgx337A Further attempts to diagnose cassette shift studies for HGCal geometry (relevant for the V18 version)

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Further attempts to diagnose cassette shift studies for HGCal geometry (relevant for the V18 version)

#### PR validation:

Use the scripts in Validation/HGCalValidation

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by srimanob** [comment]
+Upgrade

File: SimG4CMS/Calo/src/HGCalSD.cc [hunk 1]
```diff
@@ -213,10 +216,19 @@ uint32_t HGCalSD::setDetUnitId(const G4Step* aStep) {
...
-    edm::LogVerbatim(pid) << "CheckID " << HGCSiliconDetId(id) << " input position: (" << hitPoint.x() / CLHEP::cm
-                          << ", " << hitPoint.y() / CLHEP::cm << "); position from ID (" << xx << ", " << xy.second
-                          << ") distance " << diff << " Valid " << valid1 << " Wafer type|rotation " << partn.first
-                          << ":" << partn.second << " CassetteShift " << cshift;
+    int indx = HGCalWaferIndex::waferIndex(layer, hid1.waferU(), hid1.waferV());
+    double phi = std::atan2(hitPoint.y(), hitPoint.x());
+    edm::LogVerbatim(pid) << "CheckID " << HGCSiliconDetId(id) << " Layer:Module:Cell:ModuleLev " << layer << ":"
+                          << module << ":" << cell << ":" << moduleLev << " SimWt:history " << useSimWt_ << ":"
+                          << touch->GetHistoryDepth() << ":" << levelT1_ << ":" << levelT2_ << " input position: ("
+                          << hitPoint.x() / CLHEP::cm << ", " << hitPoint.y() / CLHEP::cm << ":" << convertRadToDeg(phi)
```
Comment (perrotta) [inline SimG4CMS/Calo/src/HGCalSD.cc:224]: If you don't shield the log message and related computations into some `EDM_ML_DEBUG`, you could still save some computation in normal running by letting `phi` being calculated only inside the `LogVerbatim`:
```suggestion
    edm::LogVerbatim(pid) << "CheckID " << HGCSiliconDetId(id) << " Layer:Module:Cell:ModuleLev " << layer << ":"
                          << module << ":" << cell << ":" << moduleLev << " SimWt:history " << useSimWt_ << ":"
                          << touch->GetHistoryDepth() << ":" << levelT1_ << ":" << levelT2_ << " input position: ("
                          << hitPoint.x() / CLHEP::cm << ", " << hitPoint.y() / CLHEP::cm << ":"
                          << convertRadToDeg([phi](std::atan2(hitPoint.y(), hitPoint.x())))
```

**Review by perrotta — COMMENTED** [review]
Another optimization could be applied by replacing [L213-L214](https://github.com/cms-sw/cmssw/blob/ef72c17fdc0c61a243dd0d00c5712b050024b3b4/SimG4CMS/Calo/src/HGCalSD.cc#L213-L214) by:
```C++
    double diff = dx * dx + dy * dy;
    constexpr double tol = 2.0 * 2.0;
```

**Comment by perrotta** [comment]
@bsunanda my comments above are only about some possible code optimization. Please let me know if you plan to apply them here or in a follow up PR

**Comment by bsunanda** [comment]
@perrotta I shall take care of this in a subsequent PR. Please merge this PR as it is

**Pull request merged**